pkg/logs/input/kubernetes/launcher.go

```diff
@@ -51,14 +52,27 @@ type Launcher struct {
 	serviceNameFunc    func(string, string) string // serviceNameFunc gets the service name from the tagger, it is in a separate field for testing purpose
 }
 
-// NewLauncher returns a new launcher.
-func NewLauncher(sources *config.LogSources, services *service.Services, collectAll bool) (*Launcher, error) {
+// IsAvailable retrues true if the launcher is available and a retrier otherwise
+func IsAvailable() (bool, *retry.Retrier) {
 	if !isIntegrationAvailable() {
-		return nil, fmt.Errorf("%s not found", basePath)
+		log.Errorf("Kubernetes launcher is not available. Integration not available - %s not found", basePath)
```

Comment:
If I understood correctly, if I'm running in a Docker environment, but the Docker launcher fails the first time, I will get these logs, while it's expected not to have `/var/log/pods` mounted?

Reply:
Yes - that is correct. But ideally the agent will quickly succeed and connect to docker on the next retry - if it doesn't something else is probably wrong config wise on the host.

Reply:
I see three options:
- Return based on `IsFeaturePresent`.
- Log as `info`.
- Log as `warn` only if Kubernetes/Docker is in `IsFeaturePresent`
We should not log ERR or WARN when something is working as expected

Reply:
I am probably leaning toward one of the second 2 options as the feature can be present, but kubelet connection can still fail and we want to handle those cases differently.

Reply:
If the feature is present it's legit to run the launcher and to have a warning log, however it's when the feature is not present that we don't want to log `Kubernetes launcher is not available.`